src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs

```diff
@@ -203,12 +204,46 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
             });
         }
 
-        private void Write(string message) {
-            if (CurrentWindow != null) {
-                _coreShell.DispatchOnUIThread(() => CurrentWindow.Write(message));
+        /// <summary>
+        /// Workaround for interactive window that does not currently support
+        /// 'carriage return' i.e. writing into the same line
+        /// </summary>
+        private string _lastMessage;
+        private int _lastPosition = -1;
+        private void OutputBuffer_Changed(object sender, TextContentChangedEventArgs e) {
+            if (_lastMessage != null) {
+                // Writing messages in the same line (via simulated CR)
+                var m = _lastMessage; // Make sure it is null so we don't re-enter.
+                _lastMessage = null;
+                // Replace last written placeholder with the actual message
+                CurrentWindow.OutputBuffer.Replace(new Span(_lastPosition, 1), m);
             }
         }
 
+        private void Write(string message) {
+            _coreShell.DispatchOnUIThread(() => {
+                if (CurrentWindow != null) {
+                    // If message starts with CR we remember current output buffer
+                    // length so we can continue writing lines into the same spot.
+                    // See txtProgressBar in R.
+                    if (message.StartsWithOrdinal("\r")) {
+                        // Store the message and the initial position. All subsequent 
+                        // messages that start with CR. Will be written into the same place.
+                        _lastMessage = message.Substring(1);
+                        _lastPosition = _lastPosition < 0 ? CurrentWindow.OutputBuffer.CurrentSnapshot.Length : _lastPosition;
+                        message = "|"; // replacement placeholder so we can receive 'buffer changed' event
```

Comment:
Any placeholder can be used. I am not entirely sure what would happen on ```\r``` as placeholder since generally VS editor doesn't like mixed line endings. Lonely ```\r``` is uncommon (I think) since it is just CR without LF (i.e. print on the same line).